src/pages/MapPage.js

```diff
@@ -1,3 +1,68 @@
-export default function MapPage() {
-  return <main className="MapPage Content">Map</main>;
+import "./MapPage.css";
+import { useEffect, useRef, useState } from "react";
+import RoomSelectBtn from "../components/RoomSelectBtn";
+
+export default function MapPage({ onRoomClick }) {
+  const roomsDifficulty = useRef(Array(4));
```

Comment:
Why is this not a state, but a ref?

Reply:
Well, yes I don't use a mutable value. So would it be better to use state in this case? 
I decided to use ref because I only store data in it on mount and don't want something to re-render.